libkineto/src/output_json.cpp

```diff
@@ -292,12 +293,34 @@ void ChromeTraceLogger::handleGpuActivity(
   const CUpti_ActivityKernel4* kernel = &activity.raw();
   const TraceActivity& ext = *activity.linkedActivity();
   constexpr int threads_per_warp = 32;
-  float warps_per_sm = -1.0;
+  float blocks_per_sm = -1.0;
   if (smCount_) {
-    warps_per_sm = (kernel->gridX * kernel->gridY * kernel->gridZ) *
-        (kernel->blockX * kernel->blockY * kernel->blockZ) /
-        (float) threads_per_warp / smCount_;
+    blocks_per_sm = (kernel->gridX * kernel->gridY * kernel->gridZ) / (float) smCount_;
   }
+
+  // Calculate occupancy
+  float occupancy = -1.0;
+  if (kernel->deviceId < occProps_.size()) {
```

Comment:
Please separate this out into its own function, or even better, its own file, e.g. CudaDeviceProperties.h / cpp.

Reply:
Thanks. Removed to CudaDeviceProperties.cpp